### PR TITLE
Feature/tg 117 edit profile

### PR DESCRIPTION
**Pull request by RamiroJGuillen**
# Summary

Implemented edit profile functionality and integrated with backend.

## Details

- Added a ProfileService to retrieve user information to display on profile component.
- Edited Profile Component to correctly display information retrieved instead of placeholders.
- Reutilized UserForm Component to also implement it on edit profile functionality.
- Added a validation to not show email on updating user information because email should never change.
- Implemented a validation so only an Admin can change the user's role.
- Fixed some mapping issues.

## Evidence

1. User information displaying correct information:

Screenshot 2024-03-07 at 15 40 10

2. User with the Trainer role won't be able to change their role:

Screenshot 2024-03-07 at 15 39 28

3. Admin user is able to change role:

Screenshot 2024-03-07 at 15 40 17

4. Workflow:

https://github.com/JU-DEV-LatamBootcamp/Klockan-Frontend/assets/151553728/ba81c47a-8865-4a83-96c5-28a58f3e48e9

## Known Issues
- Is necessary to change the country first so the cities select input is able to retrieve the corresponding cities of the country selected. If the select values don't change, the form will correctly send the cityId to the endpoint.
- Role isn't updated on Keycloak Realm

### CODE REVIEW

File: src/app/modules/app/pages/users/components/user-form/user-form.component.ts [hunk 1]
```diff
@@ -97,4 +106,21 @@ export class UserFormComponent implements OnInit {
...
+  private updateUser(): void {
+    const user = this.userForm.value;
+    console.log(user);
+    const formattedBirthdate = formatDate(user.birthdate);
+    user.birthdate = formattedBirthdate;
+    delete user.country;
+    console.log(user);
+    this.userService.edit(user).subscribe({
+      next: user => {
+        this.dialogRef.close(user);
```
Comment (Adrian-Mendoza-Jala) [inline src/app/modules/app/pages/users/components/user-form/user-form.component.ts:119]: Can we add Snackbar in case of success and error?

Comment (Adrian-Mendoza-Jala) [inline src/app/modules/app/pages/users/components/user-form/user-form.component.ts:117]: Can we add Snackbar in case of success and error?

Comment (RamiroJGuillen) [reply src/app/modules/app/pages/users/components/user-form/user-form.component.ts:117]: https://github.com/JU-DEV-LatamBootcamp/Klockan-Frontend/assets/151553728/9aa6f3b8-d65f-4b1d-b2e6-d3c1611abbde

is already implemented

File: src/app/modules/app/pages/users/components/user-form/user-form.component.ts [hunk 2]
```diff
@@ -97,4 +106,21 @@ export class UserFormComponent implements OnInit {
...
     });
   }
+
+  private updateUser(): void {
+    const user = this.userForm.value;
+    console.log(user);
+    const formattedBirthdate = formatDate(user.birthdate);
+    user.birthdate = formattedBirthdate;
+    delete user.country;
+    console.log(user);
```
Comment (Adrian-Mendoza-Jala) [inline src/app/modules/app/pages/users/components/user-form/user-form.component.ts]: Remove this console.log

Comment (RamiroJGuillen) [reply src/app/modules/app/pages/users/components/user-form/user-form.component.ts]: done!

**Review by Adrian-Mendoza-Jala — APPROVED** [review]

**Review by AJN-jalaU — APPROVED** [review]

**Pull request merged**